JPS_BASE_LIB/src/main/java/uk/ac/cam/cares/jps/base/slurm/job/JobSubmission.java

```diff
@@ -439,6 +443,17 @@ public void monitorJobs() throws SlurmJobException{
 				session = jsch.getSession(slurmJobProperty.getHpcServerLoginUserName(), getHpcAddress(), 22);
 				String pwd = slurmJobProperty.getHpcServerLoginUserPassword();
 				session.setPassword(pwd);
+
+				try {
```

Comment:
Logic for this block is quite unclear. It looks like restricting authentication to password based upon exception. What is the reason for that? Exceptions are usually thrown when something goes wrong. Maybe some comments would help.

Reply:
I'll add some comments.